subprojects/file-watching/src/main/java/org/gradle/internal/watch/vfs/impl/WatchingVirtualFileSystem.java

```diff
@@ -320,4 +340,67 @@ private void closeUnderLock() {
             }
         }
     }
+
+    public static class BuildStartedWithFileSystemWatchingEnabled extends WatchingBookkeepingResult {
+        private final boolean startedWatching;
+
+        public BuildStartedWithFileSystemWatchingEnabled(
+            boolean startedWatching,
+            @Nullable FileWatcherRegistry.FileWatchingStatistics fileWatchingStatistics,
+            SnapshotHierarchy vfsRoot
+        ) {
+            super(fileWatchingStatistics, vfsRoot);
+            this.startedWatching = startedWatching;
+        }
+
+        public boolean isStartedWatching() {
+            return startedWatching;
+        }
+    }
+
+    public static class BuildFinishedWithFileSystemWatchingEnabled extends WatchingBookkeepingResult {
```

Comment:
I think these types should have `Result` in their names. They are typically defined in a `BuildOperationType` subtype with generics, too, perhaps along with an (empty) `Details` object.

I'm a bit unsure about the superclass. Should the VFS retention stats be maybe a field instead?

Reply:
Both have now result in their name. I am also using delegation for the statistics.

Reply:
Should they have a "grouping" type implementing `BuildOperationType`? I think they should, if only for easy ability to find the types.

Reply:
The grouping type would be for each of them. So there would be `BuildFinishedFileSystemWatchingBuildOperationType` and `BuildStartedFileSystemWatchingBuildOperationType`.

I realized that if we don't add the `*BuildOperationType` interfaces  then we won't be able to capture the build operations in Gradle profiler, since we seem to use the [`Details` object there](https://github.com/gradle/gradle-profiler/blob/53202a9a9497d4aba243bdcaf39930cc1c9c7f1c/subprojects/build-operations/src/main/java/org/gradle/trace/buildops/BuildOperationTrace.java#L91).

I would prefer not to add them, though it seems I need to so we can measure the duration in Gradle profiler, so I'll add the 6 interfaces.